example/src/main/java/com/paystack/checkout/example/MainActivity.kt

```diff
@@ -50,12 +51,25 @@ class MainActivity : AppCompatActivity() {
                 Log.e(TAG, "Cancelled!")
             }
         }
-        val checkoutBuilder = PaystackCheckout.Builder(this, "ask4myk@gmail.com", 10000, "NGN")
+
+        val metadata = """{
+            "age": "50",
+            "custom_fields": [
+                {
+                    "display_name": "Brand",
+                    "variable_name": "brand",
+                    "value": "Bongo Kivu"
+                }
+            ]
+        }""".trimIndent()
+
+        println(metadata)
+
+        val checkoutBuilder = PaystackCheckout.Builder(this, "again@week.com",
+                50000, "NGN")
         checkoutBuilder.apply {
-            firstName("Michael")
-            lastName("Michael")
-            channels(PaymentChannel.bank, PaymentChannel.qr, PaymentChannel.card)
-            phone("07019133867")
+            reference("lezzGoLIvEagAIn41")
```

Comment:
Hey @damilola-paystack, Is there a reason the calls for setting names and channels were taken out?

Reply:
@michael-paystack the first and last names are redundant. Metadata should be used instead. The public key picks the selected channel on the dashboard. Also, we should add the optional parameters in the ReadMe file.

Reply:
I will create another PR for the updated ReadMe

Reply:
Okay. I'm thinking passing metadata isn't as type-safe as being able to set those values directly. I've always thought metadata would only be used where there are no other alternatives (e.g. custom fields).

Reply:
@michael-paystack while metadata might not be type-safe,  the lastName and firstName, if sent directly, will get dropped when it gets to our server. If the firstName and lastName are needed, the developer should call the create customer endpoint.

Reply:
Okay. It's cool. I'll merge this now and we can talk about the other stuff later.